### PR TITLE
chore: cleanup after resolution of leanprover/lean4#1926

### PR DESCRIPTION
**Pull request by kim-em**
See zulip discussion, e.g. at https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/simp.20on.20binders.

This is the fix / cleanup PR following the resolution of leanprover/lean4#1926 in leanprover/lean4#2732.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/LegendreSymbol/GaussEisensteinLemmas.lean [hunk 1]
```diff
@@ -83,7 +83,7 @@ private theorem gauss_lemma_aux₁ (p : ℕ) [Fact p.Prime] {a : ℤ}
           (∏ x in Ico 1 (p / 2).succ, if (a * x : ZMod p).val ≤ p / 2 then (1 : ZMod p) else -1) =
           ∏ x in (Ico 1 (p / 2).succ).filter fun x : ℕ => ¬(a * x : ZMod p).val ≤ p / 2, -1 :=
         prod_bij_ne_one (fun x _ _ => x)
-          (fun x => by split_ifs <;> simp_all (config := { contextual := true }))
+          (fun x => by split_ifs <;> (dsimp; simp_all))
```
Comment (eric-wieser) [inline Mathlib/NumberTheory/LegendreSymbol/GaussEisensteinLemmas.lean:86]: Why did this change?

File: Mathlib/CategoryTheory/Yoneda.lean [hunk 2]
```diff
@@ -504,4 +504,5 @@ lemma isIso_of_yoneda_map_bijective {X Y : C} (f : X ⟶ Y)
   obtain ⟨g, hg : g ≫ f = 𝟙 Y⟩ := (hf Y).2 (𝟙 Y)
   exact ⟨g, (hf _).1 (by aesop_cat), hg⟩
 
+attribute [nolint simpNF] CategoryTheory.yonedaEquiv_yoneda_map
```
Comment (mattrobball) [inline Mathlib/CategoryTheory/Yoneda.lean:507]: This can be deleted safely

**Comment by PatrickMassot** [comment]
bors d+

File: Mathlib/NumberTheory/LegendreSymbol/GaussEisensteinLemmas.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/NumberTheory/LegendreSymbol/GaussEisensteinLemmas.lean:86]: The simp call in `split_ifs` became slightly more powerful, and left us in a different state. That then resulted in `simp_all` taking a wrong turn, which can be averted by making sure `dsimp` gets a chance to beta reduce first.

Comment (eric-wieser) [reply Mathlib/NumberTheory/LegendreSymbol/GaussEisensteinLemmas.lean:86]: That suggests to me that we have a confluence issue somewhere, but I guess fixing it is not in scope here.

**Comment by semorrison** [comment]
This was rolled into #8051.

**Pull request closed without merging**